mipengine/controller/node_landscape_aggregator.py

```diff
@@ -207,70 +224,114 @@ def stop(self):
             self._keep_updating = False
             self._update_loop_thread.join()
 
+    def set_new_registy_values(self, nodes, data_models, dataset_location):
+        log_node_changes(logger, self._registries.node_registry.nodes, nodes)
+        log_data_model_changes(
+            logger,
+            self._registries.data_model_registry.data_models,
+            data_models,
+        )
+        log_dataset_changes(
+            logger,
+            self._registries.data_model_registry.dataset_location,
+            dataset_location,
+        )
+        _node_registry = NodeRegistry(nodes=nodes)
+        _data_model_registry = DataModelRegistry(
```

Comment:
`NodeRegistry` and `DataModelRegistry` are supposed to be classes that implement some functionality, nevertheless by continuously throwing the instance away and instantiating a new object you re essentially using them as DTOs